enhancements/cluster-logging/performance-tuning.md

```diff
@@ -0,0 +1,241 @@
+The only remedy is to ensure that the log collector and store can keep up with the rate of log production.
+
+#### Acknowledgements
+
+[End-to-end acknowledgement](https://vector.dev/docs/about/under-the-hood/architecture/end-to-end-acknowledgements) 
+means that source acknowledgements are _delayed_ until all relevant sinks have received the data.
+After restart, the source can re-send data that did not reach all sinks - the source acts as a persistent buffer.
+
+This is more efficient than duplicating the data again in a vector disk buffer,
+but only works for sources that support acknowledgement and "at-least-once" reliable delivery.
+
+Examples of acknowledgement sources:
+- Kafka: Kafka protocol has acknowledgements and at-least-once delivery.
+- HTTP: HTTP response can be used  to implement at-least-once.
+  Not all HTTP clients do this, but REST clients for data streaming usually do.
+- **File**: Vector's persistent "position" file can be used like an "acknowledgement" for file sources.
+  Reading from the persisted position after restart is equivalent to at-least-once delivery.
+
+#### Delivery policy implementation
+
+`Fast` policy does not provide any buffering, acknowledgement or other special configuration.
+
+`Safe` policy: can be implemented in different ways for different source/sink combinations, use the most efficient available
+- persistent sink buffer: Always for all combos, the only choice if any source does not support end-to-end ack.
+- end-to-end ack: generally more efficent if supported by all sources connected to a given sink.
+  - may interact with other options (rotate_wait, drop_oldest)
+  - may be combined with persistent buffering to give extra reliability for sources that support it.
+
+``` pseudocode
+for each `Safe` output:
+  set sink.acknowledgement=true for attached sink(s).
+  for each source of each input of each pipeline to output:
+    if source can participate in acknowledgement:
+	  set source.acknowledgement=true
+	else
+	  set output.buffer=disk
+```
+
+Persistent buffering is done to compensate for a `source` but happens on the `output`:
+- acknowledgement will still be used for sources that support it.
+- adding a buffer for the weak sources doesn not interfere with acknowledgement.
+
+NOTE: The existing `outputs[].limit` changes blocking behaviour in vector.
+Rate limits still should be enforced for `Safe` delivery.
+Review the existing limit code when implementing delivery policy so that the two work together properly.
+
+### Risks and Mitigations
+
+- Added complexity to forwarder.
+- Support cost of customers abusing or misunderstanding the parameters. \
+  For `delivery` this is mitigated by having the forward compute a "correct" config.
+- Increased customer demand for help "sizing" the logging stack: setting resources and predicting performance.
+- No new security risks are expected.
+
+### Drawbacks
+None.
+## Design Details
+
+### Open Questions
+None.
```

Comment:
I anticipate a lot of questions about the details of the abstraction to fast/safe.   Would performance benchmarking be too obscure, or helpful showing examples of the implementation and its impact on a "test cluster"?    I'm assuming QE is going to need to verify the results somehow as well.

Reply:
Check the updated text. The new text focuses on the reliability guarantees and only points out in passing that the unreliable delivery "does less bookkeeping and therefore uses less CPU/disk". I've de-emphasized the notion the trade-off is about performance, and the discussion is slated towards "You probably want reliability. If you are sure you don't then unreliable delivery *might* be a bit faster."